modules/core/src/test/java/org/apache/ignite/internal/processors/cache/query/GridCacheEmptyScanQueryTest.java

```diff
@@ -0,0 +1,152 @@
+package org.apache.ignite.internal.processors.cache.query;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+import org.apache.ignite.Ignite;
+import org.apache.ignite.IgniteCache;
+import org.apache.ignite.Ignition;
+import org.apache.ignite.cache.query.ScanQuery;
+import org.apache.ignite.configuration.CacheConfiguration;
+import org.apache.ignite.configuration.IgniteConfiguration;
+import org.apache.ignite.lang.IgniteBiPredicate;
+import org.apache.ignite.test.ignite2190.Employee;
+import org.apache.ignite.test.ignite2190.EmployeePredicate;
+import org.apache.ignite.test.ignite2190.ObjectPredicate;
+import org.apache.ignite.testframework.config.GridTestProperties;
+import org.apache.ignite.testframework.junits.common.GridCommonAbstractTest;
+import org.apache.ignite.testframework.junits.multijvm.IgniteProcessProxy;
+
+/**
+ * @author NIzhikov
+ */
+public class GridCacheEmptyScanQueryTest extends GridCommonAbstractTest {
+    /**
+     * jar name contains Employee class
+     */
+    private static final String IGNITE_2190_1_0_JAR = "ignite-2190-1.0.jar";
+
+    @Override protected boolean isMultiJvm() {
+        return true;
+    }
+
+    /** {@inheritDoc} */
+    @Override protected IgniteConfiguration getConfiguration(String gridName) throws Exception {
+        IgniteConfiguration cfg = super.getConfiguration(gridName);
+        cfg.setClientMode(false);
+        cfg.setPeerClassLoadingEnabled(true);
+        return cfg;
+    }
+
+    public void testEmptyScanQueryWithOptimizedMarshaller() throws Exception {
```

Comment:
javadoc is necessary. For example:
/**
     * @throws Exception If failed.
     */